src/Analyzer/Passes/RewriteAggregateFunctionWithIfPass.cpp

```diff
@@ -51,10 +51,15 @@ class RewriteAggregateFunctionWithIfVisitor : public InDepthQueryTreeVisitorWith
         {
             const auto & second_const_value = second_const_node->getValue();
             if (second_const_value.isNull()
-                || (lower_name == "sum" && isInt64OrUInt64FieldType(second_const_value.getType()) && second_const_value.get<UInt64>() == 0))
+                || (lower_name == "sum" && isInt64OrUInt64FieldType(second_const_value.getType()) && second_const_value.get<UInt64>() == 0
+                    && !function_node->getResultType()->isNullable()))
```

Comment:
I checking nulability of function_node result type in this case equivalent to checking `if_arguments_nodes[1]->getType()->isNullable()`?

Reply:
Yes, but that requires knowing how `if` does the casting internally, and also it'd need to be different in the next if. I decided it was simpler this way, but I could change it if preferred.

Reply:
Not a big deal, but I think for the reader it would be slightly more clear.

Reply:
Ok, seems the original approach was a better idea since it covers the case where both are const. I'll add more tests instead